cmd/habitat/community/state/transitions.go

```diff
@@ -211,3 +223,158 @@ func (t *AddNodeTransition) Validate(oldState *community.CommunityState) error {
 
 	return nil
 }
+
+type StartProcessTransition struct {
+	Process *community.Process
+}
+
+func (t *StartProcessTransition) Type() string {
+	return TransitionTypeStartProcess
+}
+
+func (t *StartProcessTransition) Patch(oldState *community.CommunityState) ([]byte, error) {
+	marshaledProcess, err := json.Marshal(t.Process)
+	if err != nil {
+		return nil, err
+	}
+
+	return []byte(fmt.Sprintf(`[{
+		"op": "add",
+		"path": "/processes/-",
+		"value": %s
+	}]`, string(marshaledProcess))), nil
+}
+
+func (t *StartProcessTransition) Validate(oldState *community.CommunityState) error {
+	// make sure no process with same process id and app name is already started
```

Comment:
why can there not be 2 processes with the same app name? i thought the name was more just for UI / description while the id was the uuid.

also does this mean there can't be 2 instances of the same app running? i feel like we may want this at some point